libraries/aws/provisioning/include/types/aws_iot_provisioning_types.h

```diff
@@ -277,6 +277,41 @@ typedef struct AwsIotProvisioningRejectedResponse
     size_t errorMessageLength;  /**< The length of the error messsage sent by the server. */
 } AwsIotProvisioningRejectedResponse_t;
 
+/**
+ * @ingroup provisioning_datatypes_paramstructs
+ * @brief Aggregates the data sent as response from AWS IoT Core for the request to create a new certificate for a CSR request.
+ *
+ * @paramfor Response Callback function of @ref provisioning_function_createkeysandcertificate
+ *
+ * The @ref AwsIotProvisioning_CreateKeysAndCertificate library API passes this object to a user-provided callback
+ * function
+ * whenever the operation completes with a response from the server.
+ */
+typedef struct AwsIotProvisioningCreateCertificateFromCsrResponse
+{
+    /** @brief The highest level HTTP based status code sent by the server. */
+    AwsIotProvisioningServerStatusCode_t statusCode;
+
+    union
```

Comment:
Generally, we have decided to minimize use of unions  in the new code unless absolutely required.  I believe there MISRA as well doxygen issue.

Reply:
Will take it as a hygiene story. Removing the union will involve a larger re-factor of the library architecture

Reply:
OK.

Reply:
If this violates MISRA, and creates problems for Doxygen, and it's new code, why not fix it now? Once the code is committed, the changes will be riskier to make.

Reply:
It is new code that fits within the library's architecture of interaction between the API and Parser modules. I agree that the union should be removed, however, removing that will require me to accept 2 separate callbacks (one for success response, and one for rejected response) from the caller, and write custom-code only for this new API.
This would be inconsistent with the existing `CreateKeysAndCertificate` and `RegisterThing` APIs. Thus, I want to tackle this hygiene re-factor in a separate PR to avoid scope creep.